openssl-dynamic/src/main/c/sslutils.c

```diff
@@ -575,6 +575,9 @@ static int SSL_CTX_setup_certs(SSL_CTX *ctx, BIO *bio, bool skipfirst, bool ca)
                 X509_free(x509);
                 return -1;
             }
+            // SSL_CTX_add_client_CA does not take ownership of the x509. It just calls X509_get_subject_name
+            // and make a duplicate of this value. So we should always free the x509 after this call.
+            X509_free(x509);
```

Comment:
+1

I'll add a link to https://github.com/netty/netty/issues/6249 and comment above the call to `SSL_CTX_add_extra_chain_cert`.